### PR TITLE
[TECH DEBT] Deprecation of ESXi in core Salt in favor of Salt VMware extensions for 3008 release

### LINKED ISSUE #62754: [TECH DEBT] Deprecate ESXi execution modules in favor of VMware Salt extensions
### Description of the tech debt to be addressed, include links and screenshots

Given these execution modules currently do not get much attention, esp. since 2016,  and all VMware support is currently occurring on the VMware extensions. Any functionality currently provided by these modules and not support by VMware extensions, should be added to VMware extensions.

Should have a single point of development for VMware support, which is currently VMware extensions.
Removes chance of confusion as to which execution module to use.

Also note that the ESXi execution modules do not fully function with currently released VMware software, since they have not been maintained sufficiently to kleep up to date with VMware API changes.

Execution Modules to deprecate:
- esxcluster.py
- esxdatacenter.py
- esxi.py
- esxvm.py

Note associated state files:
- esxcluster.py
- esxdatacenter.py
- esxi.py
- esxvm.py

These state file will need to be checked to ensure they function with VMware Salt Extensions.

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Deprecates use of ESXi and associated esxcluster, esxdatacenter, esxvm, vcenter and vsphere in core Salt, in favor of using the Salt VMware extensions.

The current core esxi,  esxcluster, esxdatacenter, esxvm etc have not been maintained by Salt for some time, and the versions of pyvmomi is circa v4/5, current version is 7, and functionality has changed and not all functionality currently works. Salt is supporting the Salt VMware extensions and this is were the ESXi functionality should be supported rather than core Salt.   This is also inline with moving current functionality to extensions.

### What issues does this PR fix or reference?
Fixes:  https://github.com/saltstack/salt/issues/62754, https://github.com/saltstack/salt/issues/62526, https://github.com/saltstack/salt/issues/62004, https://github.com/saltstack/salt/issues/61097

### New Behavior
Update Salt VMware Extensions to support required functionality, if it does not currently exist in Salt VMware extensions.  Deprecation notice allows for two major versions of Salt to have needed functionality in the VMware extensions

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by MKLeb — DISMISSED** [review]

File: salt/modules/esxcluster.py [hunk 1]
```diff
@@ -22,5 +23,27 @@ def __virtual__():
     return (False, "Must be run on a proxy minion")
 
 
+def _deprecation_message(function):
```
Comment (MKLeb) [inline salt/modules/esxcluster.py:26]: Non-blocking: some de-duping for this decorator would help, but I can't think of a place to put it 😂

Comment (dmurphy18) [reply salt/modules/esxcluster.py:26]: I stole from modules/azurearm_resources.py, used in a number of the other azurearm_xxxx.py files too, so followed that style. Liked the quick and easy style of one place to edit in a file, but yup, nice if there was a standard place for deprecation to go and them import, maybe a search for decorators to go in utils. But leave that for later.

File: changelog/62754.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Deprecate core ESXi and associated states and modules, vcenter and vsphere support in favor of Salt VMware Extensions
```
Comment (Ch3LL) [inline changelog/62754.fixed:1]: The file extension should be `.deprecated` since you are deprecating modules.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: changelog/62754.fixed (hunk 2 above)
Comment (dmurphy18) [reply changelog/62754.fixed:1]: Thanks for the catch, fixed it

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
merging since tests were passing before editing changelog

**Pull request merged**